core/server/data/schema/schema.js

```diff
@@ -300,5 +300,12 @@ module.exports = {
         created_by: {type: 'string', maxlength: 24, nullable: false},
         updated_at: {type: 'dateTime', nullable: true},
         updated_by: {type: 'string', maxlength: 24, nullable: true}
+    },
+    sessions: {
+        id: {type: 'string', maxlength: 32, nullable: false, primary: true},
```

Comment:
`express-session` generates ids of length 32, however we can pass in our own id generation function, so i can change this to 24 and use ObjectIDs?

Reply:
IMO i would prefer ObjectID's for the `id` property, otherwise it's confusing. But let me know if you think we should use the express session id for the `id` property. Maybe it makes sense and i don't have the insights :)

Reply:
The only thing with ObjectID is it not very secure it uses `Math.random` and `Date.now` as entropy. We sign our cookies so an attacker would still need the secret for that, but if that was compromised - guessing a session id would be a lot easier if it was an ObjectID

Reply:
My thinking was more like: the `session id` is independent from the ObjectId.

e.g. 
```
sessions:
  id: String
  session_id: String
```

The `id` represents the actual object. The `session_id` represents the session. But i agree, maybe it's not super useful, but i don't know. It's edgy.

Reply:
I don't know 🤔 How does the session id ensures it's unique?

Reply:
It uses: https://www.npmjs.com/package/uid-safe

Reply:
> It uses: https://www.npmjs.com/package/uid-safe

This is using crypto.randomBytes. Random Bytes is strong, but is it that reliable and unique to use it for a database identifier? 🤔 

When we have decided to switch to ObjectId's as an identifier for a resource, we investigated a long time if that is doable. ObjectId's are unique because they 4 components to create the id ([see](https://github.com/williamkapke/bson-objectid/blob/master/objectid.js#L194)). And the same concept is used in mongo. Read [here](https://github.com/TryGhost/Ghost/issues/7494) through the switch to ObjectId's.

Let's take an example:

Let's imagine we list all active sessions in the admin client.
We add an endpoint `GET /sessions/` and `GET /sessions/:id/`. 
We won't expose the session key here? But we could expose the id of the resource where we store the session information.

Reply:
> Let's imagine we list all active sessions in the admin client.
> We add an endpoint GET /sessions/ and GET /sessions/:id/.
> We won't expose the session key here? But we could expose the id of the resource where we store the session information.

+1 I will sort this out for you tomorrow